terraform/modules/opensearch_domain/opensearch.tf

```diff
@@ -0,0 +1,29 @@
+resource "aws_opensearch_domain" "opensearch" {
+  domain_name    = var.domain_name
+  engine_version = var.engine
+
+  advanced_security_options {
+    internal_user_database_enabled = true
+    master_user_options {
+      master_user_name     = var.master_user_name
+      master_user_password = var.master_user_password
+    }
+  }
+
+  cluster_config {
```

Comment:
We need to support these variables in this block:

- [dedicated_master_count](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain#dedicated_master_count)
- [dedicated_master_enabled](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain#dedicated_master_enabled)
- [dedicated_master_type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain#dedicated_master_type)
- [instance_count](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain#instance_count)